alacritty/src/display/window.rs

```diff
@@ -302,7 +302,8 @@ impl Window {
 
         WinitWindow::default_attributes()
             .with_decorations(window_config.decorations != Decorations::None)
-            .with_window_icon(icon.ok())
+            .with_window_icon(icon.as_ref().ok().cloned())
```

Comment:
How does `as_ref().ok().cloned()` make sense? If anything shouldn't just cloning this early remove the need for `as_ref`?

Reply:
You can not clone unless you map to `Option`, because `Error` here is not `Clone`, so you need to `ok()` and then clone, so it should be a ref to prevent move.